src/openeo_aggregator/backend.py

```diff
@@ -672,14 +677,60 @@ def _get_connection_and_backend_service_id(
 
     def service_types(self) -> dict:
         """https://openeo.org/documentation/1.0/developers/api/reference.html#operation/list-service-types"""
-        # TODO: add caching. Also see https://github.com/Open-EO/openeo-aggregator/issues/78#issuecomment-1326180557
+        cached_info = self._memoizer.get_or_call(key=("_get_service_types",), callback=self._get_service_types)
```

Comment:
The cache key doesn't need a leading underscore (and doesn't have to be a tuple), just `"service_types"`  is good enough for now

Reply:
Corrected the name.  And indeed it should not have been a tuple either.